### PR TITLE
fix: replacing subdomain and hostname with domain

### PR DESCRIPTION
**Pull request by russorat**
updates docs to remove subdomain and hostname, replaces them with domain.

### CODE REVIEW

File: docs/integrations/teams/webhooks.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 To integrate Microsoft Teams webhooks with ngrok:
 1. [Launch your local webhook.](#start-your-app) `npm start`
-1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --hostname myexample.ngrok.io`
+1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --domain myexample.ngrok.io`
```
Comment (natasha-jarus) [inline docs/integrations/teams/webhooks.md:8]: should use `ngrok.dev` here? and elsewhere in this file?

File: docs/integrations/whatsapp/webhooks.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ description: Develop and test WhatsApp webhooks from localhost
 
 To integrate WhatsApp webhooks with ngrok:
 1. [Launch your local webhook.](#start-your-app) `npm run startFacebook`
-1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --hostname myexample.ngrok.io`
+1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --domain myexample.ngrok.io`
```
Comment (natasha-jarus) [inline docs/integrations/whatsapp/webhooks.md:12]: ditto on `ngrok.dev` in this file

File: docs/ngrok-agent/changelog.md [hunk 3]
```diff
@@ -127,12 +127,10 @@ Starts a tunnel with a random subdomain, for example a URL like `http://d95211d2
...
+Add a `domain` property with the same name as the tunnel:
 
     tunnels:
       webapp:
       proto: http
       addr: 80
-      subdomain: webapp
-
-This behavior changed in order to make it possible to launch tunnels with random domains. This was preventing the use of the configuration file and agent API to free tier users.
+      domain: webapp.ngrok.io
```
Comment (natasha-jarus) [inline docs/ngrok-agent/changelog.md:136]: `ngrok.dev`

**Review by natasha-jarus — APPROVED** [review]
:+1:  thanks for doing this! left a couple of notes about `ngrok.io` domains, but that's it :)

File: docs/integrations/facebook/webhooks.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ description: Develop and test Facebook webhooks from localhost
 
 To integrate Facebook webhooks with ngrok:
 1. [Launch your local webhook.](#start-your-app) `npm run startFacebook`
-1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --hostname myexample.ngrok.io`
+1. [Launch ngrok.](#start-ngrok) `ngrok http 3000 --region us --domain myexample.ngrok.dev`
```
Comment (natasha-jarus) [inline docs/integrations/facebook/webhooks.md:12]: can drop `--region` here

File: docs/integrations/facebook/webhooks.md [hunk 5]
```diff
@@ -70,15 +70,15 @@ Once your app is running successfully on localhost, let's get it on the internet
 
 1. On the **Domains** page, click **+ Create Domain** or **+ New Domain**.
 
-1. In the **Domain** pane, select a **Region** for your domain (i.e. `United States`), provide a value for the **Domain** field (i.e. `myexample.ngrok.io`), and then click **Continue**.
+1. In the **Domain** pane, select a **Region** for your domain (i.e. `United States`), provide a value for the **Domain** field (i.e. `myexample.ngrok.dev`), and then click **Continue**.
```
Comment (natasha-jarus) [inline docs/integrations/facebook/webhooks.md:73]: ditto on region here; these domains shouldn't be regional

File: docs/integrations/whatsapp/webhooks.md [hunk 6]
```diff
@@ -70,15 +70,15 @@ Once your app is running successfully on localhost, let's get it on the internet
 
 1. On the **Domains** page, click **+ Create Domain** or **+ New Domain**.
 
-1. In the **Domain** pane, select a **Region** for your domain (i.e. `United States`), provide a value for the **Domain** field (i.e. `myexample.ngrok.io`), and then click **Continue**.
+1. In the **Domain** pane, select a **Region** for your domain (i.e. `United States`), provide a value for the **Domain** field (i.e. `myexample.ngrok.dev`), and then click **Continue**.
```
Comment (natasha-jarus) [inline docs/integrations/whatsapp/webhooks.md:73]: more regions!

File: docs/secure-tunnels/tunnels/http-tunnels.md [hunk 7]
```diff
@@ -27,18 +27,18 @@ Anyone who can guess your tunnel URL can access your local web server unless you
...
     ![](/img/docs/cname.png)
-3.  Create a DNS CNAME record from `dev.example.com` to your CNAME target. In this example, we would point the CNAME record to `2w9c34maz.cname.ngrok.io`
-4.  Invoke ngrok with the `--hostname` switch and specify the name of your custom domain as an argument. Make sure the `--region` you specify matches the region in which you reserved your domain.
+3.  Create a DNS CNAME record from `dev.example.com` to your CNAME target. In this example, we would point the CNAME record to `2w9c34maz.wdv31sd.ngrok-cname.com`
+4.  Invoke ngrok with the `--domain` switch and specify the name of your custom domain as an argument. Make sure the `--region` you specify matches the region in which you reserved your domain.
     
     ###### Example: Run a tunnel over a custom domain
     
-        ngrok http --region=us --hostname=dev.example.com 8000
+        ngrok http --region=us --domain=dev.example.com 8000
```
Comment (natasha-jarus) [inline docs/secure-tunnels/tunnels/http-tunnels.md:41]: rrrrrrrrrrrrrrrrrrregion, also see above line 37

File: docs/secure-tunnels/tunnels/tls-tunnels.md [hunk 8]
```diff
@@ -11,31 +11,31 @@ Check out the [ngrok agent TLS tunnel documentation](/ngrok-agent/ngrok#ngrok-tl
...
+    ngrok tls --region=us --domain=secure.example.com 443
 
 ## Terminating TLS connections {#tls-termination}
 
 It's possible that the service you're trying to expose may not have the capability to terminate TLS connections. The ngrok agent can do this for you so that you can encrypt your traffic end-to-end but not have to worry about whether the local service has TLS support. Specify both the `-crt` and `-key` command line options to specify the filesystem paths to your TLS certificate and key and the ngrok agent will take care of terminating TLS connections for you.
 
 ###### Offload TLS Termination to the ngrok Agent
 
-    ngrok tls --region=us --hostname secure.example.com --key /path/to/tls.key --crt /path/to/tls.crt 80
+    ngrok tls --region=us --domain secure.example.com --key /path/to/tls.key --crt /path/to/tls.crt 80
```
Comment (natasha-jarus) [inline docs/secure-tunnels/tunnels/tls-tunnels.md:38]: region :pensive:

**Review by natasha-jarus — APPROVED** [review]
:+1: thank you for tolerating my nitpicking :)

File: docs/platform/pops.md [hunk 9]
```diff
@@ -29,12 +29,12 @@ To explicitly choose the region, add the `--region` flag or set the `region` pro
...
+TCP addresses are allocated for a specific region (the US region by default). When you reserve a TCP address, you must select a target region. You may not bind a TCP address reserved in another region other than the one it was allocated for. Attempting to do so will yield an error and prevent your tunnel session from initializing.
 
 ### Limitations {#limits}
 
 An ngrok agent may only be connected a single region, meaning you cannot host tunnels in multiple regions simultaneously with the same agent. You can run multiple ngrok agents if you need to do this.
 
-A domain cannot be reserved for multiple regions at the same time. It is not possible to geo-balance DNS to the same tunnel name in multiple regions. Use region-specific subdomains or TLDs if you need to do this.
-
-An ngrok Edge can only contain endpoints in the same region.
+It is not possible to geo-balance DNS to the same tunnel name in multiple regions. Use region-specific subdomains or TLDs if you need to do this.
```
Comment (russorat) [inline docs/platform/pops.md:40]: @natasha-jarus is this still accurate?

Comment (natasha-jarus) [reply docs/platform/pops.md:40]: i am not sure what a tunnel name is? but i do know:

- for agent initiated tunnels (`ngrok http`, `ngrok tcp`, &c), you cannot have several tunnels in different regions serving the same hostport
- for labeled tunnels, it is entirely possible to have several tunnels with the same labels in different regions all serving the same hostport.

**Comment by natasha-jarus** [comment]
@nijikokun with russ out, do you think you'd be willing to take a look at the last remaining question here and get this into a mergeable state?

File: docs/platform/pops.md (hunk 9 above)
Comment (nijikokun) [reply docs/platform/pops.md:40]: Would be good to point to edges for this use-case :+1:

Comment (nijikokun) [inline docs/platform/pops.md:40]:
```suggestion
To geo-balance DNS to the same tunnel name in multiple regions you can use the ngrok Cloud Edges feature or use region-specific subdomains or TLDs.
```

**Review by nijikokun — APPROVED** [review]
Looks good, found one additional mention of `tunnel.ngrok-agent` and pushed up a fix for it.

**Review by jrobsonchase — APPROVED** [review]

**Pull request merged**